src/osd/OSDMap.cc

```diff
@@ -4265,6 +4267,43 @@ void print_osd_utilization(const OSDMap& osdmap,
   }
 }
 
+unsigned OSDMap::get_pool_suggested_min_size(const pg_pool_t &pool) const
+{
+  unsigned suggested_min_size = 0;
+  switch (pool.get_type()) {
+  case pg_pool_t::TYPE_REPLICATED:
+    suggested_min_size = pool.get_size() - pool.get_size() / 2;
+    break;
+  case pg_pool_t::TYPE_ERASURE:
+    {
+      ErasureCodeInterfaceRef erasure_code;
+      ostringstream ss;
+      ErasureCodeProfile profile =
+        get_erasure_code_profile(pool.erasure_code_profile);
+      ErasureCodeProfile::const_iterator plugin = profile.find("plugin");
+      if (plugin == profile.end()) {
+        break;
+      }
+      ErasureCodePluginRegistry &instance =
+        ErasureCodePluginRegistry::instance();
+      int err = instance.factory(plugin->second,
+                                 g_conf->get_val<std::string>("erasure_code_dir"),
+                                 profile,
+                                 &erasure_code,
+                                 &ss);
```

Comment:
I'm not wild about loading up the plugin every time we're doing this health check.  Maybe we can cache it somewhere?